frigate/ffmpeg_presets.py

```diff
@@ -138,27 +184,54 @@ def parse_preset_hardware_acceleration_decode(arg: Any) -> list[str]:
     return PRESETS_HW_ACCEL_DECODE.get(arg, None)
 
 
+def _parse_rotation_scale(
+    arg: Any,
+    mode: str,
+    rotate: int,
+) -> str:
+    """Return the correct rotation scale or "" if preset none is set."""
+    if not isinstance(arg, str) or " " in arg:
+        return ""
+
+    if rotate == 90:
+        transpose = "clock"
+    elif rotate == 180:
+        if arg.startswith("preset-vaapi") or arg.startswith("preset-intel-qsv"):
+            transpose = "reverse"
+        else: # No 'reverse' option suported, then 2 'clocks' rotations
+            transpose =  "clock,transpose=clock"
+    elif rotate == 270:
+        transpose =  "cclock"
+    else : # Rotation not need or not supported
+        return ""
+
+    return PRESETS_HW_ACCEL_SCALE_ROTATION.get(arg, "").get(mode).format(transpose)
+
+
 def parse_preset_hardware_acceleration_scale(
     arg: Any,
     detect_args: list[str],
     fps: int,
     width: int,
     height: int,
+    rotate: int,
 ) -> list[str]:
     """Return the correct scaling preset or default preset if none is set."""
     if not isinstance(arg, str) or " " in arg:
         scale = PRESETS_HW_ACCEL_SCALE["default"].format(fps, width, height).split(" ")
         scale.extend(detect_args)
         return scale
 
+    transpose =_parse_rotation_scale(arg, "detect", rotate)
+
     scale = PRESETS_HW_ACCEL_SCALE.get(arg, "")
 
     if scale:
-        scale = scale.format(fps, width, height).split(" ")
+        scale = scale.format(fps, width, height, transpose).split(" ")
```

Comment:
Seems like this is going to lead to multiple -vf args, I think this logic needs to be refactored. 

We're also going to need more logic for the vf formatting as many of these are going to need hwupload/hwdownload args

Reply:
This is still an issue I think, the scale and rotation both have -vf which will cause issues